### DISCUSSION TITLE
Kanboard - mysql automatic actions not working.

### DISCUSSION THREAD

**Discussion by hankbkk** [kanboard/kanboard#5201] [Installation & Configuration]
Kanboard - v1.2.18
Mysql - Ver 8.0.23
Ubuntu - 20.04.1
Cronjob list for user www-data :  * * * * * cd /var/www/html/kanboard && ./cli cronjob >/dev/null 2>&1
/var/www/html/kanboard permission :  drwxrwxr-x 8 www-data www-data  4096 Apr 19 11:45 kanboard

Automatic Action configured :
 Move the task to another column when assigned to a user
    Event name = Task assignee change
    Source column = New Case
    Destination column = Work in progress

Test case:
1) create new task and then edit that task by assigning it to a user.
2) this should move the task to 'Work In Progress' lane, but task remains in "New Case" lane.

Any help is much appreciated.

Regards,
Hank.

**Comment by rafacamargo123** [comment]
please change the cronjob from `./cli cronjob >/dev/null 2>&1` to `./cli cronjob >/var/www/html/kanboard/cronjob.log 2>&1` then paste here the content of `/var/www/html/kanboard/cronjob.log` after a cronjob execution.

**Comment by hankbkk** [comment]
hi Rafacamargo123,

this is the current crontab for user www-data
0 8 * * * cd /var/www/html/kanboard && ./cli cronjob >/var/www/html/kanboard/cronjob.log 2>&1

I've tried creating a task and tested the automatic action but no cronjob was created, no log was created.
Please tell me how may i test this cronjob for the automatic action.

My automatic action is :

Move the task to another column when assigned to a user
      Event name = Task assignee change
      Source column = New Task
      Destination column = Work in progress

I created a new task and then edited that task to assign a staff, but it didn't move that task to the next column.

Much appreciate your help. Thanks

**Comment by rafacamargo123** [comment]
Hi!

Your crontab is set to run only at 8:00 am:

Hour minute day month weekday

Change it to:

```
* * * * * cd /var/www/html/kanboard && ./cli cronjob >/var/www/html/kanboard/cronjob.log 2>&1
```
To make sure it will run every minute.

But wait, this kind of automatic action you’re using doesn’t depend on the cronjob. 

Are you using any plugin?

**Comment by hankbkk** [comment]
Hi Rafacamargo123,

Yes, not sure on why that cronjob was required, i just followed the installation instruction.

I have the following Plug-ins installed.
![image](https://user-images.githubusercontent.com/82799711/116495191-a631ac00-a8cc-11eb-8e14-fb272116fe9c.png)

I have created :
https://kanboard.discourse.group/t/kanboard-mysql-automatic-actions-not-working/1785
and will close this once we are in that forum.

Thanks very much for your help.

**Comment by rafacamargo123** [comment]
Okay, I'll try to reproduce your issue with the same plugins, because I cannot reproduce on a fresh install.
Ill answer you there.
The cronjob is needed for other actions/tasks that are time-based such as "move task when is expired", "notificate when expired", etc.
The installation procedure is to set the cronjob to run once a day (like `0 8 * * *` for 8 AM every day), but I prefeer to keep it `* * * * *` (every minute) so the tasks are expired immediately. The problem is that if you enable notifications for expired tasks, You'll be notified everytime the cronjob runs and the tasks is still expired.
So, knowing this, you can choose to keep the cronjob daily or change it to every minute, every hour, etc.

But this exact problem is not related to cronjobs.

I'll answer you at the forum.

**Comment by rafacamargo123** [comment]
I was able to reproduce this issue on a fresh install, without additional plugins:

If "Event name = Task assignee change":
_This information is actually incorrect. See next comment for further details._
The action only works if the task is assigned to someone, and then you change the assignee to another user
If the task is unassigned, when you change to someone, the task doesn't run.
If you use the "Assign to me" shortcut, then the action runs, no matter who was the last assignee.

If "Event name = Task modification":
The action runs if you change anything on a task, anything at all, AND if assignee is <> "".
In this case, the action runs even if you don't change the assignee.

**Comment by rafacamargo123** [comment]
I finally understood why my testing were getting different results:

If you **bulk create tasks**, then the action will not run for the tasks that you bulk-created if you only change the assignee.

Step by step reproduction:

1. Bulk create one or more tasks (unassigned)
2. Set an assignee on one of the tasks

Result: action won't run.
It will only run if you change something first, then change the assignee on the next edit.

If you create a task using the creation form, it will run as expected.

What I said about the "Event name = Task modification" is true. That's the actual behaviour.

The behaviour for "Event name = Task assignee change" is actually the following:

If you edit something on a task together with the assignee field, the action doesn't run.
The event "Task assignee change" is only fired if **the only change in the task is the assignee field**.

If you edit the title + assignee, for instance, the action will not execute.

So, there's a couple of issues here:

- first assignee edition on bulk created tasks do not fire the "Task assignee change" event.
- the "Task assignee change" is only fired if the only modification made is on the assignee field.

Probably those two issues are related. I need to test, but I think bulk created tasks are being created without some sort of information, that is being setted when you first edit the task, adding stuff to the task changes, and making the assignee change not unique. Then, the event is not fired because another modification was made with the assignee change.

@hankbkk can you verify what I said above?

- Does the action run if you edit a task and change ONLY the assignee field?
- Does the action run if you use the "Assign to me" shortcut?

**Comment by hankbkk** [comment]
Hi rafacamargo123,

Extremely sorry for the long silence. One of my staff was diagnosed with Covid, so we were all denied access to the office. 

Answers to your questions :
1)  Does the action run if you edit a task and change ONLY the assignee field?
*** I created a task and assigned it to a user. Then i opened that task again and changed the assignee to another staff, but the automatic action still didn't execute.

2) Does the action run if you use the "Assign to me" shortcut?
*** Yes, the automatic action works if i use the "Assign to me" shortcut.

3) i am not using bulk creation as this is all individual tickets created by different staff. During creation, the assignee would not be assigned yet, only in rare cases where the ticket is created by the assignee himself.  

I really appreciate your help with this Rafacamargo123, wish i can return the favour when i can.
